packages/react/src/utils/renderComponent.tsx

```diff
@@ -109,12 +109,18 @@ const renderComponent = <P extends {}>(
     : {}
 
   // Resolve styles using resolved variables, merge results, allow props.styles to override
-  const mergedStyles: ComponentSlotStylesPrepared = mergeComponentStyles(
-    theme.componentStyles[displayName],
-    withDebugId({ root: props.design }, 'props.design'),
-    withDebugId({ root: props.styles }, 'props.styles'),
-    withDebugId({ root: animationCSSProp }, 'props.animation'),
-  )
+  let mergedStyles: ComponentSlotStylesPrepared = theme.componentStyles[displayName] || {
+    root: () => ({}),
+  }
+
+  if (!_.isNil(props.design) || !_.isNil(props.animation) || !_.isNil(props.styles)) {
```

Comment:
If there are inline overrides, then these will be merged on top of the theme, and it would be more expensive to use them..